File: src/main/java/me/miki/shindo/management/mods/impl/Items2DMod.java
package me.miki.shindo.management.mods.impl;

import me.miki.shindo.Shindo;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.management.mods.Mod;
import me.miki.shindo.management.mods.ModCategory;
import me.miki.shindo.management.notification.NotificationType;

public class Items2DMod extends Mod {

    private static Items2DMod instance;

    public Items2DMod() {
        super(TranslateText.ITEMS_2D, TranslateText.ITEMS_2D_DESCRIPTION, ModCategory.RENDER);

        instance = this;
    }

    public static Items2DMod getInstance() {
        return instance;
    }

    @Override
    public void onEnable() {
        super.onEnable();

        if (ItemPhysicsMod.getInstance().isToggled()) {
            ItemPhysicsMod.getInstance().setToggled(false);
            Shindo.getInstance().getNotificationManager().post(TranslateText.ITEM_PHYSICS.getText(), "Disabled due to incompatibility", NotificationType.WARNING);
        }
    }
}

File: src/main/kotlin/me/miki/shindo/management/mods/impl/Items2DMod.kt
package me.miki.shindo.management.mods.impl

import me.miki.shindo.Shindo
import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.management.mods.Mod
import me.miki.shindo.management.mods.ModCategory
import me.miki.shindo.management.nanovg.font.Lucide
import me.miki.shindo.management.nanovg.font.Shinconic
import me.miki.shindo.management.notification.NotificationType

class Items2DMod : Mod(TranslateText.ITEMS_2D, TranslateText.ITEMS_2D_DESCRIPTION, ModCategory.RENDER, Shinconic.MOD_ITEMS2_D) {
    init {
        instance = this
    }

    override fun onEnable() {
        super.onEnable()

        val itemPhysics = ItemPhysicsMod.instance
        if (itemPhysics != null && itemPhysics.isToggled()) {
            itemPhysics.setToggled(false)
            Shindo.getInstance().getNotificationManager().post(
                TranslateText.ITEM_PHYSICS.getText(),
                "Disabled due to incompatibility",
                NotificationType.WARNING,
            )
        }
    }

    companion object {
        @JvmField
        var instance: Items2DMod? = null
    }
}
